doc/source/toolchain.rst

```diff
@@ -0,0 +1,226 @@
+ Date             Pythons supported
+================  =======================================================================
+ 2018              Py2.7, Py3.4+ (SciPy 1.2.x is the last release to support Python 2.7)
+ 2019              Py3.5+ (but Py2.7-specific code not removed)
+ 2020              Py3.5+ (removal of Py2.7-specific code permitted)
+================  =======================================================================
+
+NumPy
+^^^^^
+
+SciPy depends on NumPy but releases of SciPy are not tied to releases of NumPy.
+SciPy attempts to be compatible with at least the 4 previous releases of NumPy.
+In particular, SciPy can not rely on features of just the latest NumPy, but needs to be
+written using what is common in all of those 4 releases. [1]_, [6]_.
+
+========  ========================    ===========================
+ Python    Minimum NumPy version       Maximum NumPy version
+========  ========================    ===========================
+3.7         1.13.1                     >= 1.16.x
+3.6         1.12.1                     >= 1.16.x
+3.5         1.9.3                      >= 1.16.x
+2.7         1.8.2                      1.16.x
+========  ========================    ===========================
+
+
+C Compilers
+^^^^^^^^^^^
+
+SciPy is compatible with most modern C compilers.  However CPython on Windows is
+built with specific versions of the Microsoft Visual C++ compiler [3]_, as is the
+corresponding build of SciPy.  This has implications for the C language standards
+that can be supported.
+
+===================   ==============   ===================
+CPython               MS Visual C++    C Standard
+===================   ==============   ===================
+2.7, 3.0, 3.1, 3.2       9.0           C90
+3.3, 3.4                10.0           C90 & some of C99
+3.5, 3.6                14.0           C90 & most of C99
+===================   ==============   ===================
+
+
+
+C and C++ Language Standards
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+C and C++ language standards for SciPy are generally guidelines
+rather than official decisions. This is particularly true of
+attempting to predict adoption timelines for newer standards.
+
+================  ===========================================
+ Date              C Standard
+================  ===========================================
+ <= 2018           C90
+ 2019              C90 for old code, may consider C99 for new
+ 2020              C99
+ ?                 C11
+ ?                 C17, C18
+================  ===========================================
+
+C99 has been a standard for 20 years, but experience has shown that
+not all features are supported equally well across all platforms.
+C18 is a bug fix for C11, so C11 may be skipped entirely.
+
+For C++, we currently suggest that anything beyond C++11 is going to be impossible
+for a very long time because of ecosystem support restrictions. See [4]_.
```

Comment:
How about something like: in practice the C++ feature set that can be used is limited by what the MSVC versions we need to use support. C++11 can be used, C++14/17 is going to be ....

Reply:
Done.